src/Juvix/Core/MainLang.hs

```diff
@@ -265,12 +265,10 @@ cType ii g (Pa pi varType resultType) ann = undefined
 cType ii g (NPm first second) ann = undefined
 -- (Lam) introduction rule of dependent function type
 -- Lam s should be of dependent function type (Pi pi ty ty').
-cType ii g (Lam s) (sig, VPi pi _ty ty') = do
-  traceShowM s
-  let sVal = cEval s []
+cType ii g (Lam s) (sig, (VPi pi ty ty')) =
   cType
     (ii + 1)
-    ((Local ii, (sig <.> pi, sVal)) : g) -- put s in the context with usage sig*pi
+    ((Local ii, (sig <.> pi, ty)) : g) -- put s in the context with usage sig*pi
```

Comment:
You are correct. I should have put the ```ty``` (instead of the evaluated ```s```) in the context. This is a quick fix that I will merge in so that the tests pass. In the future we'll all make smaller, separate PR so we can merge them with more flexibility.